src/main/java/pl/newsler/components/user/UserCrudService.java

```diff
@@ -12,15 +10,15 @@
 import pl.newsler.commons.models.NLPassword;
 import pl.newsler.commons.models.NLSecretKey;
 import pl.newsler.commons.models.NLSmtpAccount;
-import pl.newsler.commons.models.NLType;
+import pl.newsler.commons.models.NLUserType;
 import pl.newsler.security.NLIPasswordEncoder;
 
 import java.util.Optional;
 import java.util.UUID;
 
 @RequiredArgsConstructor
 class UserCrudService implements IUserCrudService {
-    private final UserRepository userRepository;
+    private final IUserRepository userRepository;
     private final NLIPasswordEncoder passwordEncoder;
 
     @Override
```

Comment:
Annotate param with `@NotNull` from validation package. Use `NLModel.validate()` to chcek its correctness.